src/Orc.Automation.Tests/NUnit/Attributes/ScenarioAttribute.cs

```diff
@@ -0,0 +1,70 @@
+﻿#nullable enable
+namespace Orc.Automation.Tests;
+
+using System;
+using System.Linq;
+using NUnit.Framework.Interfaces;
+using NUnit.Framework.Internal.Commands;
+using NUnit.Framework;
+using NUnit.Framework.Internal;
+using ScenarioManagement;
+
+[AttributeUsage(AttributeTargets.Method)]
+public class ScenarioTestCaseAttribute : TestCaseAttribute, IWrapTestMethod
+{
+    private readonly string _scenarioName;
+
+    public ScenarioTestCaseAttribute(string scenarioName, params object[] arguments)
+        : base(arguments)
+    {
+        _scenarioName = scenarioName;
+    }
+
+    public string? ScenarioSuite { get; set; }
+
+    public TestCommand Wrap(TestCommand command)
+    {
+        const string scenarioAlreadyRunKey = "ScenarioAlreadyRun";
+
+        var testProperties = command.Test.Parent?.Properties;
+        if (testProperties?.ContainsKey(scenarioAlreadyRunKey) == true)
+        {
+            return command;
+        }
+
+        var description = string.Empty;
+        var loggers = Array.Empty<IAutomationScenarioLogger>();
+
+        if (testProperties is not null)
+        {
+            if (testProperties.ContainsKey(PropertyNames.Description))
+            {
+                description = testProperties[PropertyNames.Description]
+                    .OfType<string>()
+                    .FirstOrDefault() ?? string.Empty;
+            }
+
+            var logScenarioKey = typeof(LogScenarioAttribute<>).Name;
+            if (testProperties.ContainsKey(logScenarioKey))
```

Comment:
TryGet?

Reply:
it's not a dictionary... it's NUnit.Framework.Interfaces.IPropertyBag... it doesn't have TryGet